homeassistant/components/hisense_aehw4a1/climate.py

```diff
@@ -0,0 +1,415 @@
+    HVAC_MODE_OFF,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_ONLY,
+]
+
+FAN_MODES = [
+    "mute",
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_HIGH,
+    FAN_AUTO,
+]
+
+SWING_MODES = [
+    SWING_OFF,
+    SWING_VERTICAL,
+    SWING_HORIZONTAL,
+    SWING_BOTH,
+]
+
+PRESET_MODES = [
+    PRESET_NONE,
+    PRESET_ECO,
+    PRESET_BOOST,
+    PRESET_SLEEP,
+    "sleep_2",
+    "sleep_3",
+    "sleep_4",
+]
+
+AC_TO_HA_STATE = {
+    "0001": HVAC_MODE_HEAT,
+    "0010": HVAC_MODE_COOL,
+    "0011": HVAC_MODE_DRY,
+    "0000": HVAC_MODE_FAN_ONLY,
+}
+
+HA_STATE_TO_AC = {
+    HVAC_MODE_OFF: "off",
+    HVAC_MODE_HEAT: "mode_heat",
+    HVAC_MODE_COOL: "mode_cool",
+    HVAC_MODE_DRY: "mode_dry",
+    HVAC_MODE_FAN_ONLY: "mode_fan",
+}
+
+AC_TO_HA_FAN_MODES = {
+    "00000000": FAN_AUTO,  # fan value for heat mode
+    "00000001": FAN_AUTO,
+    "00000010": "mute",
+    "00000100": FAN_LOW,
+    "00000110": FAN_MEDIUM,
+    "00001000": FAN_HIGH,
+}
+
+HA_FAN_MODES_TO_AC = {
+    "mute": "speed_mute",
+    FAN_LOW: "speed_low",
+    FAN_MEDIUM: "speed_med",
+    FAN_HIGH: "speed_max",
+    FAN_AUTO: "speed_auto",
+}
+
+AC_TO_HA_SWING = {
+    "00": SWING_OFF,
+    "10": SWING_VERTICAL,
+    "01": SWING_HORIZONTAL,
+    "11": SWING_BOTH,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _build_entity(device):
+    _LOGGER.debug("Found device at %s", device)
+    return ClimateAehW4a1(device)
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up the AEH-W4A1 climate platform."""
+    # Priority 1: manual config
+    if hass.data[DOMAIN].get(CONF_IP_ADDRESS):
+        devices = hass.data[DOMAIN].get(CONF_IP_ADDRESS, [])
+    else:
+        # Priority 2: scanned interfaces
+        devices = await AehW4a1().discovery()
+
+    all_devices = [_build_entity(device) for device in devices]
+    async_add_devices(all_devices, True)
+
+
+class ClimateAehW4a1(ClimateDevice):
+    """Representation of a Hisense AEH-W4A1 module for climate device."""
+
+    def __init__(self, device):
+        """Initialize the climate device."""
+        self._unique_id = device
+        self._device = AehW4a1(device)
+        self._hvac_modes = HVAC_MODES
+        self._fan_modes = FAN_MODES
+        self._swing_modes = SWING_MODES
+        self._preset_modes = PRESET_MODES
+        self._available = None
+        self._on = None
+        self._temperature_unit = None
+        self._current_temperature = None
+        self._target_temperature = None
+        self._hvac_mode = None
+        self._fan_mode = None
+        self._swing_mode = None
+        self._preset_mode = None
+        self._previous_state = None
+
+    async def async_update(self):
+        """Pull state from AEH-W4A1."""
+        try:
+            status = await self._device.command("status_102_0")
+        except Exception as library_error:  # pylint: disable=broad-except
+            _LOGGER.debug("Unexpected error: %s", library_error)
```

Comment:
Depending on the exception, consider raising the log level. We want a single log message when an entity switches between available and unavailable, usually at warning level.